api/envoy/extensions/filters/network/local_ratelimit/v3/local_rate_limit.proto

```diff
@@ -51,4 +54,9 @@ message LocalRateLimit {
   // and ``share_key`` configuration. All fields of ``token_bucket`` must match exactly for the token bucket to be shared. If this
   // field is empty, this filter will not share a token bucket with any other filter.
   string share_key = 4;
+
+  // Optional delay configuration for rejecting the connection after some specified time duration
+  // instead of immediately rejecting the connection. That way, a malicious user is not able to
+  // retry as fast as possible which provides a better DoS protection for Envoy.
+  google.protobuf.Duration delay = 5;
```

Comment:
thanks for your comments! Looks like WAF is not implemented in envoy yet. I agree that this approach has some limitations, but it might be an alternative/workaround before WAF is supported. And I can revise the field name